community/graph-algo/src/main/java/org/neo4j/graphalgo/GraphAlgoFactory.java

```diff
@@ -354,6 +360,51 @@
     }
 
     /**
+     * See {@link #dijkstra(PathExpander, CostEvaluator)}
+     *
+     * Instead of finding all shortest paths with equal cost, find the top {@param numberOfWantedPaths} paths.
+     * This is usually slower than finding all shortest paths with equal cost.
+     *
+     * Uses a cost evaluator which uses the supplied property key to
+     * represent the cost (values of type <bold>double</bold>).
+     *
+     * @param expander the {@link PathExpander} to use for expanding
+     * {@link Relationship}s for each {@link Path}.
+     * @param relationshipPropertyRepresentingCost the property to represent cost
+     * on each relationship the algorithm traverses.
+     * @param numberOfWantedPaths number of paths to find.
+     * @return an algorithm which finds the cheapest path between two nodes
+     * using the Dijkstra algorithm.
+     */
+    public static PathFinder<WeightedPath> dijkstra( PathExpander expander,
+            String relationshipPropertyRepresentingCost, int numberOfWantedPaths )
+    {
+        return dijkstra( expander, new DoubleEvaluator( relationshipPropertyRepresentingCost ), numberOfWantedPaths );
+    }
+
+    /**
+     * See {@link #dijkstra(PathExpander, CostEvaluator)}
+     *
+     * Instead of finding all shortest paths with equal cost, find the top {@param numberOfWantedPaths} paths.
+     * This is usually slower than finding all shortest paths with equal cost.
+     *
+     * @param expander the {@link PathExpander} to use for expanding
+     * {@link Relationship}s for each {@link Path}.
+     * @param costEvaluator evaluator that can return the cost represented
+     * by each relationship the algorithm traverses.
+     * @param numberOfWantedPaths number of paths to find.
+     * @return an algorithm which finds the cheapest path between two nodes
+     * using the Dijkstra algorithm.
+     */
+    public static PathFinder<WeightedPath> dijkstra( PathExpander expander,
+            CostEvaluator<Double> costEvaluator, int numberOfWantedPaths )
+    {
+        return new Dijkstra( expander, costEvaluator, Precision.EPSILON,
+                PathInterestFactory.numberOfShortest( Precision.EPSILON, numberOfWantedPaths ) );
+    }
+
+    /**
+     * @deprecated Dijkstra should not be used with state on {@link PathExpander}
```

Comment:
when you deprecate it would be nice to add a link to what you are suppose to use instead